api_tests/comments/test_views.py

```diff
@@ -0,0 +1,1037 @@
+        assert_equal(urlparse(url).path, expected_url)
+
+    def test_comment_has_node_link(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url)
+        url = res.json['data']['relationships']['node']['links']['related']['href']
+        expected_url = '/{}nodes/{}/'.format(API_BASE, self.public_project._id)
+        assert_equal(urlparse(url).path, expected_url)
+
+    def test_comment_has_target_link_with_correct_type(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url)
+        url = res.json['data']['relationships']['target']['links']['related']['href']
+        expected_url = '/{}nodes/{}/'.format(API_BASE, self.public_project._id)
+        target_type = res.json['data']['relationships']['target']['links']['related']['meta']['type']
+        expected_type = 'node'
+        assert_equal(urlparse(url).path, expected_url)
+        assert_equal(target_type, expected_type)
+
+    def test_comment_has_replies_link(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url)
+        url = res.json['data']['relationships']['replies']['links']['self']['href']
+        expected_url = '/{}comments/{}/replies/'.format(API_BASE, self.public_comment)
+        assert_equal(urlparse(url).path, expected_url)
+
+    def test_comment_has_reports_link(self):
+        self._set_up_public_project_with_comment()
+        res = self.app.get(self.public_url)
+        url = res.json['data']['relationships']['reports']['links']['related']['href']
+        expected_url = '/{}comments/{}/reports/'.format(API_BASE, self.public_comment)
+        assert_equal(urlparse(url).path, expected_url)
+
+    def test_private_node_only_logged_in_contributor_commenter_can_update_comment(self):
+        self._set_up_private_project_with_comment()
+        res = self.app.put_json_api(self.private_url, self.payload, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(self.payload['data']['attributes']['content'], res.json['data']['attributes']['content'])
+
+    def test_private_node_logged_in_non_contributor_cannot_update_comment(self):
+        self._set_up_private_project_with_comment()
+        res = self.app.put_json_api(self.private_url, self.payload, auth=self.non_contributor.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_private_node_logged_out_user_cannot_update_comment(self):
+        self._set_up_private_project_with_comment()
+        res = self.app.put_json_api(self.private_url, self.payload, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def test_public_node_only_contributor_commenter_can_update_comment(self):
+        self._set_up_public_project_with_comment()
+
+        # Contributor who made the comment
+        res = self.app.put_json_api(self.public_url, self.public_comment_payload, auth=self.user.auth)
```

Comment:
This (and many other tests in the file) have more than one scenario per test; can you break these out for the reasons discussed in other comments?

Rather than annotate the same thing over and over, I'll just make this a blanket remark.